File: backend/app/schemas/trend.py
import uuid
from datetime import datetime

from pydantic import BaseModel


class TrendFetchLogOut(BaseModel):
    id: uuid.UUID
    requested_at: datetime
    woeid: int
    success: bool
    status_code: int | None
    error_detail: str | None
    raw_response: dict | None
    trend_count: int

    model_config = {"from_attributes": True}


class TrendFetchLogListResponse(BaseModel):
    items: list[TrendFetchLogOut]
    total: int
    page: int
    limit: int


class TrendOut(BaseModel):
    id: uuid.UUID
    name: str
    tweet_volume: int | None
    woeid: int
    fetched_at: datetime
    times_used: int

    model_config = {"from_attributes": True}


class TrendListResponse(BaseModel):
    items: list[TrendOut]
    total: int
    page: int
    limit: int

File: frontend/src/types/trend.ts
export interface TrendFetchLog {
  id: string
  requested_at: string
  woeid: number
  success: boolean
  status_code: number | null
  error_detail: string | null
  raw_response: Record<string, unknown> | null
  trend_count: number
}

export interface TrendFetchLogListResponse {
  items: TrendFetchLog[]
  total: number
  page: number
  limit: number
}

export interface Trend {
  id: string
  name: string
  tweet_volume: number | null
  woeid: number
  fetched_at: string
  times_used: number
}

export interface TrendListResponse {
  items: Trend[]
  total: number
  page: number
  limit: number
}
